core/src/dsnp/pseudo_relationship_identifier.rs

```diff
@@ -0,0 +1,63 @@
+use crate::dsnp::dsnp_types::DsnpPrid;
+use crypto_box::{aead::AeadInPlace, Nonce, PublicKey, SalsaBox, SecretKey};
+use hkdf::Hkdf;
+use sha2::Sha256;
+use x25519_dalek::x25519;
+
+const PRI_CONTEXT: &[u8] = b"PRIdCtx0";
+struct PseudonymousRelationshipIdentifier;
+
+impl PseudonymousRelationshipIdentifier {
+	fn create(a: u64, b: u64, a_secret_key: &SecretKey, b_public_key: &PublicKey) -> DsnpPrid {
+		let id_b = b.to_le_bytes();
+
+		let root_shared = x25519(*a_secret_key.as_bytes(), *b_public_key.as_bytes());
+		let hk = Hkdf::<Sha256>::new(Some(&id_b[..]), &root_shared);
+		let mut derived_key = [0u8; 64];
+		hk.expand(PRI_CONTEXT, &mut derived_key).unwrap();
+
+		let mut public_key_arr = [0u8; 32];
+		let mut secret_key_arr = [0u8; 32];
+		public_key_arr.copy_from_slice(&derived_key[..32]);
+		secret_key_arr.copy_from_slice(&derived_key[32..]);
+		let salsa = SalsaBox::new(&public_key_arr.into(), &secret_key_arr.into());
```

Comment:
can be replaced with `XSalsa20Poly1305::new(&key);` if willing to use that library directly